File: tests/errors/optional.swift
public enum BasicError: Error {
	case requested
}

public func attempt(shouldThrow: Bool) throws -> Bool {
	if shouldThrow {
		throw BasicError.requested
	}
	return true
}

public func recover(shouldThrow: Bool) -> Bool? {
	return try? attempt(shouldThrow: shouldThrow)
}

File: tests/errors/optional.js
export function attempt$shouldThrow$(shouldThrow) {
  if (shouldThrow) {
    throw 0;
  }

  return true;
}
export function recover$shouldThrow$(shouldThrow) {
  let $try;

  try {
    $try = attempt$shouldThrow$(shouldThrow);
  } catch (e) {
    $try = null;
  }

  return $try;
}